src/dg/dg.cpp

```diff
@@ -3330,6 +3330,839 @@ real2 DGBase<dim,real,MeshType>::discontinuity_sensor(
+        std::array<std::vector<real>,nstate> soln_dofs;
+
+        const unsigned int n_shape_fns = n_dofs_curr_cell / nstate;
+        std::array<real, nstate> local_min;
+        for(unsigned int istate=0; istate<nstate; ++istate){
+            local_min[istate] = 1e9;
+
+            //allocate soln_dofs
+            soln_dofs[istate].resize(n_shape_fns);
+        }
+
+        for(unsigned int idof = 0; idof<n_dofs_curr_cell; ++idof){
+            const unsigned int istate = this->fe_collection[poly_degree].system_to_component_index(idof).first;
+            const unsigned int ishape = this->fe_collection[poly_degree].system_to_component_index(idof).second;
+            soln_dofs[istate][ishape] = this->solution[current_dofs_indices[idof]]; //
+
+            if(soln_dofs[istate][ishape] < local_min[istate])
+                local_min[istate] = soln_dofs[istate][ishape];
+        }
+
+        const unsigned int n_quad_pts = this->volume_quadrature_collection[poly_degree].size();
+        const std::vector<real> &quad_weights = this->volume_quadrature_collection[poly_degree].get_weights();
+        //interpolate solution dofs to quadrature pts.
+        //and apply integral for the soln avg
+        std::array<real, nstate> soln_cell_avg = {};
+
+        std::array<std::vector<real>,nstate> soln_at_q;
+
+        for(int istate=0; istate<nstate; istate++){
+            soln_at_q[istate].resize(n_quad_pts);
+            soln_basis.matrix_vector_mult_1D(soln_dofs[istate], soln_at_q[istate],
+                                             soln_basis.oneD_vol_operator);
+        }
+
+        for(unsigned int istate=0; istate<nstate; ++istate){
+            for(unsigned int iquad = 0; iquad<n_quad_pts; ++iquad)
+            {
+                if (flux_nodes_type==FluxNodes::GL)
+                { 
+                   if (soln_at_q[istate][iquad] < local_min[istate])
+                        local_min[istate] = soln_at_q[istate][iquad];
+                }
+
+                soln_cell_avg[istate] += quad_weights[iquad]
+                    * soln_at_q[istate][iquad];
+            }
+        }
+
+        real eps = 1e-13;
+        real gamma = 1.4;
+
+        if(nstate == PHILIP_DIM + 2)
+        {
+            // compute value of pressure at soln_cell_avg
+            real rho_avg = soln_cell_avg[0];
+            real tot_energy_avg = soln_cell_avg[nstate - 1];
+            real m_avg = soln_cell_avg[1];
+
+
+            real p_avg = (gamma-1) * (tot_energy_avg - 0.5 *(((m_avg*m_avg)/rho_avg)));
+            if (dim > 1)
+                p_avg += -0.5*((soln_cell_avg[2]*soln_cell_avg[2])/rho_avg);
+            if (dim > 2)
+                p_avg += -0.5*((soln_cell_avg[3]*soln_cell_avg[3])/rho_avg);
+
+            //get epsilon (lower bound for density) for theta limiter
+            eps = std::min({ all_parameters->pos_eps, rho_avg, p_avg });
+            if (eps < 0) eps = all_parameters->pos_eps;
+        }
```

Comment:
All this is physic logic. If you need logic such as computing pressure, provide a Physics object that can do so (like Euler, Navier-Stokes). We should not be redefining equations here or specifying gamma here.

Reply:
With the new abstract class for the bound_preserving_limiters, a pointer is created for Euler Physics which is now used to compute the pressure within the positivity preserving limiters.